bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    hbaseSnapshotDir = cloudTestDataFolder + "data/";
+    UUID test_uuid = UUID.randomUUID();
+    restoreDir = cloudTestDataFolder + "restore/" + test_uuid;
+
+    // Cloud Storage config
+    GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
+    gcpOptions.setProject(projectId);
+    gcsUtil = new GcsUtil.GcsUtilFactory().create(gcpOptions);
+
+    // Bigtable config
+    connection = BigtableConfiguration.connect(projectId, instanceId);
+    tableId = "test_" + UUID.randomUUID().toString();
+
+    System.out.println("Setting up integration tests");
+
+    String[] keys = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+    keySplits = new byte[keys.length][];
+    for (int i = 0; i < keys.length; i++) {
+      keySplits[i] = keys[i].getBytes();
+    }
+  }
+
+  private static String getTestProperty(String name) {
+    return checkNotNull(System.getProperty(name), "Required property missing: " + name);
+  }
+
+  @After
+  public void teardown() throws IOException {
+    final List<GcsPath> paths = gcsUtil.expand(GcsPath.fromUri(restoreDir + "/*"));
+
+    if (!paths.isEmpty()) {
+      final List<String> pathStrs = new ArrayList<>();
+
+      for (GcsPath path : paths) {
+        pathStrs.add(path.toString());
+      }
+      this.gcsUtil.remove(pathStrs);
+    }
+
+    connection.close();
+
+    // delete test table
+    BigtableConfiguration.connect(projectId, instanceId)
+        .getAdmin()
+        .deleteTable(TableName.valueOf(tableId));
+  }
+
+  @Test
+  public void testHBaseSnapshotImport() throws Exception {
+
+    try (Connection connection = BigtableConfiguration.connect(projectId, instanceId)) {
+      // Crete table
+      System.out.println("DEBUG (create test table) ==>");
+      TableName tableName = TableName.valueOf(tableId);
+      HTableDescriptor descriptor = new HTableDescriptor(tableName);
+
+      descriptor.addFamily(new HColumnDescriptor(CF));
+
+      connection.getAdmin().createTable(descriptor, SnapshotTestingUtils.getSplitKeys());
+
+      // Start import
+      System.out.println("DEBUG (import snapshot) ==>");
+      DataflowPipelineOptions importPipelineOpts =
+          PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+      importPipelineOpts.setRunner(DataflowRunner.class);
+      importPipelineOpts.setGcpTempLocation(dataflowStagingLocation);
+      importPipelineOpts.setNumWorkers(1);
+      importPipelineOpts.setProject(projectId);
+      importPipelineOpts.setRegion(region);
+
+      ImportJobFromHbaseSnapshot.ImportOptions importOpts =
+          importPipelineOpts.as(ImportJobFromHbaseSnapshot.ImportOptions.class);
+      // setup GCP and bigtable
+      importOpts.setBigtableProject(StaticValueProvider.of(projectId));
+      importOpts.setBigtableInstanceId(StaticValueProvider.of(instanceId));
+      importOpts.setBigtableTableId(StaticValueProvider.of(tableId));
+      importOpts.setBigtableAppProfileId(null);
```

Comment:
why is this necessary?

Reply:
It is not, removed, thanks!

Reply:
This was copied from https://github.com/googleapis/java-bigtable-hbase/blob/486589b9361bf0e62b466b332bbc23964ffe4d0c/bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java#L198 with some confusing comment so I kept it.

After some testing, it is not needed, will remove this.